networkx/algorithms/isomorphism/tree_isomorphism.py

```diff
@@ -1,279 +1,686 @@
-    # rename the nodes with consecutive integers
-    # so that all nodes get a unique name between both trees
+def get_initial_maps_and_height(T, root):
+    """Get the initial maps and height used by the rooted tree isomorphism alg.
+
+    Define and return the following maps: LEVELS, VALUES, PARENTHOOD and
+    CHILDREN. Also return the rooted tree's height.
+
+    For the LEVELS map. Let h be the height of the tree. The amount of
+    levels a tree has is equal to its height. If a vertex v is on the i-th
+    level, then its childs are on the (i-1)-th level. Thus, by definition, the
+    only vertex in the h-th level is the root of the tree and the only vertices
+    on the level 0 are leaves. Define a LEVELS map such that, for each i in
+    {0,...,h}, LEVELS(i) is the set of vertices found on the i-th level of T.
+
+    For the VALUES map. A value of a vertex corresponds to the structure it has
+    on the rooted tree. Informally, the structure of a vertex can be thought as
+    the structure or "form" the sub-tree rooted on v has. Define the initial
+    values of the VALUES map such that, for every vertex v in the rooted tree,
+    if v is a leaf then v's initial value is 0, otherwise its defined as -1.
+
+    The PARENTHOOD map is the parent-child relationship obtained when traversing
+    the rooted tree, in this case the algorithm used is BFS.
+
+    For the CHILDREN map. Defines a map CHILDREN where CHILDREN(v) is the
+    ordered list of children of v in the rooted tree. This map determines who
+    are the parents of the leaves found on the 0-th level.
+
+    Parameters
+    ----------
+    T : NetworkX Graph
+        The rooted tree.
+
+    root : node
+        The root of the tree.
+
+    Returns
+    -------
+    LEVELS : dict of int: set of nodes
+        A LEVELS map such that, for each i in {0,...,h}, LEVELS(i) is the set of
+        vertices found on the i-th level of T.
+
+    VALUES : dict of node: int
+        A VALUES map such that, for each vertex v in the rooted tree, if v is a
+        leaf then v's initial value is 0, otherwise its defined as -1.
+
+    CHILDREN : dict of node: list of nodes
+        A CHILDREN map such that CHILDREN(v) is the ordered list of children
+        of v in the rooted tree. This map determines who are the parents of the
+        leaves found on the 0-th level.
+
+    PARENTHOOD : dict of node: node
+        The parenthood map of the rooted tree.
+
+    height : int
+        The height of the rooted tree.
+
+    Note
+    ----
+    This algorithm runs in O(n) time and space.
 
-    # our new "fake" root node is 0
-    # t1 is numbers from 1 ... n
-    # t2 is numbered from n+1 to 2n
     """
+    # Define an empty map for LEVELS, VALUES, CHILDREN and PARENTHOOD.
+    LEVELS = defaultdict(set)
+    CHILDREN = defaultdict(list)
+    VALUES = {}
+    PARENTHOOD = {}
 
-    dT = nx.DiGraph()
+    # Auxiliary variables.
+    from_distance = defaultdict(set)
+    distance_to_root = {}
```

Comment:
You don't need this variable (or the 3 below it) ; at the end of the BFS, the height is simply `len(from_distance) - 1` !